src/awkward/operations/structure.py

```diff
@@ -2157,6 +2157,202 @@ def getfunction(layout, depth, posaxis):
         return out
 
 
+def _packed(array, axis=None, highlevel=True, behavior=None):
+    layout = ak.operations.convert.to_layout(
+        array, allow_record=False, allow_other=False
+    )
+    nplike = ak.nplike.of(layout)
+
+    def truncate(layout, n):
+        return ak.layout.IndexedArray64(
+            ak.layout.Index64(nplike.arange(n)), layout
+        ).project()
+
+    def apply(layout, depth, posaxis):
+        if posaxis is not None:
+            # If a particular axis was given
+            posaxis = layout.axis_wrap_if_negative(posaxis)
+            # Do not proceed past that axis
+            if posaxis < depth - 1:
+                return layout
+
+        if isinstance(layout, ak.layout.NumpyArray):
+            return layout.contiguous()
+
+        # EmptyArray is a no-op
+        if isinstance(layout, ak.layout.EmptyArray):
+            return layout
```

Comment:
All the `if` statements at this level should be an `elif` chain (i.e. avoid accidental fall-through if one handler is missing a `return` statement). It's just a more "functional" way of doing things, and a little safer. You'll notice that it's the same pattern in all the other places where we check all node types.